src/api/app/models/user.rb

```diff
@@ -75,6 +75,13 @@ class User < ApplicationRecord
   scope :in_beta, -> { where(in_beta: true) }
   scope :in_rollout, -> { where(in_rollout: true) }
 
+  scope :watchlisted_requests, ->(user) { BsRequest.joins(:watched_items).where(watched_items: { user: user }).order('number DESC') }
```

Comment:
Hm no need to join monstrous tables if you can just fetch the IDs of the `BsRequest`s from `watched_items`?

Reply:
This is updated. We are using `watched_items` association here